internal/applier/helm.go

```diff
@@ -90,6 +90,11 @@ func (h *Helm) Apply(ctx context.Context, contentFS fs.FS, ext *ocv1alpha1.Clust
 		}
 	}
 
+	// NOTE: We need to be cautious of eating errors here.
+	// We should always return any error that occurs during an
+	// attempt to install/upgrade/reconcile. Only when there is
+	// a verifiable reason to eat the error (i.e it is recoverable)
+	// should an exception be made.
```

Comment:
I think we should call out here the specific error that cannot be eaten. Something like:

```
The following kinds of errors should be returned up the stack to ensure exponential backoff can occur:
- Permission errors (it is not possible to watch changes to permissions. The only way to eventually recover from permission errors is to keep retrying).
```

Reply:
Also, I think it is _probably_ safe to delete this instance of the comment, and leave just the one in `clusterextension_controller.go`.

My reasoning there is that I think it is unlikely that we would swallow errors here if we didn't have a way to handle them well. In the reconcile code, I could easily see us swallowing errors after updating our status object with their contents.

Reply:
Updated the comments in https://github.com/operator-framework/operator-controller/pull/1104/commits/97ce5d773526bb19691a8bf4ae1330b72e039350